ami/interactions/environments/audio_observation_generator.py

```diff
@@ -0,0 +1,181 @@
+import math
+from pathlib import Path
+from typing import Iterator, TypeAlias
+
+import soundfile as sf
+import torch
+import torchaudio.functional as F
+
+StrPath: TypeAlias = str | Path
+
+
+class ChunkedStridedAudioReader:
+    """Audio file reader that supports chunked reading with stride.
+
+    This class reads audio files in chunks of specified size with a
+    given stride, optionally performing resampling to a target sample
+    rate.
+    """
+
+    def __init__(
+        self,
+        audio_file: StrPath,
+        chunk_size: int,
+        stride: int,
+        target_sample_rate: int | None = None,
```

Comment:
```suggestion
        sample_rate: int,
```
`sample_rate`でも十分伝わりそうなのと、サンプルレートって学習や検証で重要なハイパラだし、configや初期化のコード見た時一目で把握できるようにしたい　ということでOptionalじゃなく必須にしたい

Reply:
修正: 
https://github.com/MLShukai/ami/pull/271/commits/bb063a44f0abfcb3b92848f6a9e34368143ac532